src/plugins/plugin.gridLines.js

```diff
@@ -0,0 +1,282 @@
+				borderDashOffset = helpers.getValueOrDefault(gridLines.borderDashOffset, globalDefaults.borderDashOffset);
+			}
+
+			// Get position of current gridLine
+			var x1, x2, y1, y2;
+			if (isHorizontal) {
+				var xLineValue = scale.getPixelForTick(index);
+				xLineValue += helpers.aliasPixel(lineWidth);
+
+				x1 = x2 = xLineValue;
+				y1 = chartArea.top;
+				y2 = chartArea.bottom;
+			} else {
+				var yLineValue = scale.getPixelForTick(index) + helpers.aliasPixel(lineWidth);
+
+				x1 = chartArea.left;
+				x2 = chartArea.right;
+				y1 = y2 = yLineValue;
+			}
+
+			// First and last gridLine of first found axis for each direction are marked as undefined borders.
+			// First and last gridLines are never drawn for any scale to ensure that there won't be any overlapping on chartArea borders.
+			// All four sides of chartArea have to be marked, to be sure there will always be some border. If the border is defined by
+			// any axis, that border will be preferred
+			if (index === 0) {
+				// The undefinedBorder variables are used to determine if the undefinedBorders for the
+				// specific direction is already set, as every direction can only have them set once.
+				if (isHorizontal && undefinedBorderOptions.horizontal === undefined) {
+					undefinedBorderOptions.horizontal = {
+						lineWidth: gridLines.lineWidth,
+						lineColor: gridLines.lineColor,
+						borderDash: gridLines.borderDash,
+						borderDashOffset: gridLines.borderDashOffset
+					};
+				} else if (!isHorizontal && undefinedBorderOptions.vertical === undefined) {
+					undefinedBorderOptions.vertical = {
+						lineWidth: gridLines.lineWidth,
+						lineColor: gridLines.lineColor,
+						borderDash: gridLines.borderDash,
+						borderDashOffset: gridLines.borderDashOffset
+					};
+				} else {
+					continue;
+				}
+
+				// Add first gridLine of this scale as an undefined border
+				bordersToDraw.push(getUndefinedBorder(index, scale, chartArea, x1, x2, y1, y2));
+
+				// Add last gridLine of this scale as an undefined border
+				bordersToDraw.push(getUndefinedBorder(gridLinesCount-1, scale, chartArea, x1, x2, y1, y2));
+
+			} else if (index !== gridLinesCount-1) {
+				context.lineWidth = lineWidth;
+				context.strokeStyle = lineColor;
+				if (context.setLineDash) {
+					context.setLineDash(borderDash);
+					context.lineDashOffset = borderDashOffset;
+				}
+
+				// Draw current gridLine
+				drawLine(context, x1, x2, y1, y2);
+			}
+		}
+	}
+
+	function drawBorders(context, bordersToDraw, undefinedBorderOptions) {
+		// Draws all the borders. Skips undefined orders which would be overlapped by a defined border
+		helpers.each(bordersToDraw, function(borderToDraw) {
+			// When the border is undefined, check if there isn't a defined border on the same place, if there is one dont draw this border
+			if (!borderToDraw.undefinedBorder || bordersToDraw.findIndex(bordersOverlap, borderToDraw) === -1) {
```

Comment:
I realize this might not have good performance, but I think it's fine given that the number of axes is usually small